src/shared/serverinfo.hpp

```diff
@@ -0,0 +1,231 @@
+#ifndef SERVERINFO_HPP
+#define SERVERINFO_HPP
+
+#include <vector>
+#include <string>
+
+#include "defs.hpp"
+
+struct serverinfo;
+
+// list of known servers, with or without informations
+// TODO:
+//   * should be a private static of serverinfo
+//   * stop using raw pointers (RAII ftw)
+extern vector<serverinfo*> servers;
+
+// code imported, here to reduce build dependencies
+extern bool filterstring(char *dst, const char *src, bool newline, bool colour, bool whitespace, bool wsstrip, size_t len);
+namespace client
+{
+    extern bool serverinfocompare(serverinfo const*a, serverinfo const*b);
+}
+
+namespace server
+{
+    extern int getver( int );
+}
+
+//this enum lists (some of) the server properties that can be used for sorting.
+//The C and C++ specs says and enum starts with the value 0 except if explicitly
+//specified.
+//It have 2 list of members:
+// * row 1: original names, for compat, and maybe to allow some naming support in
+//          CS latter
+// * row 2: aliases to avoid reapeating 'SINFO::' for nothing.
+// TODO: use scoped enums
+enum SINFO
+{
+    SINFO_NONE   ,
+    SINFO_STATUS ,
+    SINFO_NAME   ,
+    SINFO_PORT   ,
+    SINFO_QPORT  ,
+    SINFO_DESC   ,
+    SINFO_MODE   ,
+    SINFO_MUTS   ,
+    SINFO_MAP    ,
+    SINFO_TIME   ,
+    SINFO_NUMPLRS,
+    SINFO_MAXPLRS,
+    SINFO_PING   ,
+    SINFO_PRIO   ,
+    SINFO_MAX    ,
+};
+
+// statuses a server can have, about being joinable
+// TODO: use scoped enums
+enum SSTAT
+{
+    SSTAT_OPEN   ,
+    SSTAT_LOCKED ,
+    SSTAT_PRIVATE,
+    SSTAT_FULL   ,
+    SSTAT_UNKNOWN,
+    SSTAT_MAX    ,
+};
+
+// values returned when trying to join a server?
+// used in m_attr[5]
+enum MM
+{
+    MM_OPEN    ,
+    MM_VETO    ,
+    MM_LOCKED  ,
+    MM_PRIVATE ,
+    MM_PASSWORD,
+};
+
+struct serverinfo
+{
+    // should probably be private, but:
+    // WAITING is used in serverbrowser for cubescript exposition
+    enum
+    {
+        MAXPINGS = 3,
+        WAITING = INT_MAX
+    };
+
+private:
+    // list of player/spectators names
+    std::vector<std::string> m_players;
```

Comment:
TODO: discuss whether to use any form of prefix. There's a lot of pros and cons against them. I've used all kinds of prefixes (and also omitted them entirely), and I have no clear preference. In my own upcoming PR I used a simple trailing underscore...